### PR TITLE
Add new diagram and more detailed steps description

### PR DESCRIPTION
**Pull request by butenkor**
The idea is to give a short but better overview on the integration process without diving to deep into the details of the integration guide.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,15 +19,14 @@
...
-3. The commercetools platform communicates with the extension module.
-4. The extension module communicates with the Adyen payment provider.
-5. Adyen returns a payment result to the extension module.
-6. The extension module updates the commercetools payment.
-7. The shop verifies the updated payment.
-8. The shop presents the results to the shopper.
+![Payment flow](./docs/images/payment-flow.svg)
+1. Front end uses [Adyen Web Coponents](https://docs.adyen.com/checkout/supported-payment-methods) to present required payment methods. The list of available payment methods can be also obtained through the integration with the help of [get available payment methods request](./extension/docs/WebComponentsIntegrationGuide.md#step-3-get-available-payment-methods-optional). On user interaction, frontend web-component generates JSON payloads like [make payment](https://docs.adyen.com/payment-methods/cards/web-component#make-a-payment) or submit additional payment data which has to be provided to commercetools payment as described [here](./extension/docs/WebComponentsIntegrationGuide.md#web-components-integration-guide).
```
Comment (leungkinghin) [inline README.md:23]: > or submit additional payment data which has to be provided to 

Could we provide a reference here to link with the additional payment section in integration guide?

Comment (butenkor) [reply README.md:23]: I did link it with our guide under word [here] or you mean something else?

Comment (lojzatran) [inline README.md:23]:
```suggestion
1. Front end uses [Adyen Web Coponents](https://docs.adyen.com/checkout/supported-payment-methods) to present required payment methods. The list of available payment methods can be also obtained through the integration with the help of [get available payment methods request](./extension/docs/WebComponentsIntegrationGuide.md#step-3-get-available-payment-methods-optional). On user interaction, frontend web-component generates JSON payloads like [make payment](https://docs.adyen.com/online-payments/components-web#step-3-make-a-payment) or [submit additional payment data](https://docs.adyen.com/online-payments/components-web#step-5-additional-payment-details) which has to be provided to commercetools payment as described [here](./extension/docs/WebComponentsIntegrationGuide.md#web-components-integration-guide).  
```

File: README.md [hunk 2]
```diff
@@ -19,15 +19,14 @@
...
-4. The extension module communicates with the Adyen payment provider.
-5. Adyen returns a payment result to the extension module.
-6. The extension module updates the commercetools payment.
-7. The shop verifies the updated payment.
-8. The shop presents the results to the shopper.
+![Payment flow](./docs/images/payment-flow.svg)
+1. Front end uses [Adyen Web Coponents](https://docs.adyen.com/checkout/supported-payment-methods) to present required payment methods. The list of available payment methods can be also obtained through the integration with the help of [get available payment methods request](./extension/docs/WebComponentsIntegrationGuide.md#step-3-get-available-payment-methods-optional). On user interaction, frontend web-component generates JSON payloads like [make payment](https://docs.adyen.com/payment-methods/cards/web-component#make-a-payment) or submit additional payment data which has to be provided to commercetools payment as described [here](./extension/docs/WebComponentsIntegrationGuide.md#web-components-integration-guide).  
+2. With help of the API-Extension commercetools sends provided data to the Extension Module.
```
Comment (ahmetoz) [inline README.md:24]:
```suggestion
2. The commercetools platform sends the payment data to the publicly exposed Extension Module with the help of [commercetools HTTP API Extensions](https://docs.commercetools.com/http-api-projects-api-extensions). 
```

File: README.md [hunk 3]
```diff
@@ -19,15 +19,14 @@
...
+3. The Extension Module processes provided data by the front end, exchanges it with Adyen and provides synchronous response back to the front end / commercetools caller. Based on the result the front end either creates an order or continues with further payment steps as described [here](./extension/docs/WebComponentsIntegrationGuide.md#web-components-integration-guide).
```
Comment (ahmetoz) [inline README.md:25]:
```suggestion
3. The Extension Module processes provided web component payload passed by the front end, exchanges it with Adyen API, and provides synchronous response back to the front end / commercetools caller. Based on result, the front end either creates an order or continues with further payment steps as described at [integration guide](./extension/docs/WebComponentsIntegrationGuide.md#web-components-integration-guide).
```
> provides synchronous response back to the front end / commercetools caller.

I am not sure about this part. Is the extension module response back to the front end?

> the front end either creates an order

Does the front end create an order? => I guess for simplicity you are considering the merchant server as the front end ?

**Review by ahmetoz — COMMENTED** [review]
Overall looks good.

File: README.md (hunk 1 above)
Comment (leungkinghin) [reply README.md:23]: Lam has addressed it :)

File: README.md (hunk 3 above)
Comment (butenkor) [reply README.md]: > I am not sure about this part. Is the extension module response back to the front end?

I try to keep it as simple as possible and follow what one sees on the diagram - without not going to much into details.

> Does the front end create an order? => I guess for simplicity you are considering the merchant server as the front end ?

For simplicity there is no merchant server on the diagram but front end is :)

File: README.md (hunk 2 above)
Comment (butenkor) [reply README.md:24]: What about this version as "provided data" is also in use in the previous step so there is a link by wording already? What is an extension and how it works can be checked by following the link so I would keep it short.
```suggestion
2. With help of the [commercetools HTTP API Extensions](https://docs.commercetools.com/http-api-projects-api-extensions) provided data is sent to the Extension Module.
```

File: README.md (hunk 1 above)
Comment (butenkor) [reply README.md]: ok Thanks. I commited Lams suggestion.

**Review by ahmetoz — APPROVED** [review]

**Review by lojzatran — APPROVED** [review]

**Pull request merged**